test/services/database/challenge_repository_test.dart

```diff
@@ -0,0 +1,206 @@
+import "package:cloud_firestore/cloud_firestore.dart";
+import "package:fake_cloud_firestore/fake_cloud_firestore.dart";
+import "package:flutter_test/flutter_test.dart";
+import "package:proxima/models/database/challenge/challenge_data.dart";
+import "package:proxima/models/database/challenge/challenge_firestore.dart";
+import "package:proxima/models/database/post/post_data.dart";
+import "package:proxima/models/database/post/post_firestore.dart";
+import "package:proxima/models/database/post/post_id_firestore.dart";
+import "package:proxima/models/database/user/user_firestore.dart";
+import "package:proxima/services/database/challenge_repository_service.dart";
+import "package:proxima/services/database/post_repository_service.dart";
+
+import "../../mocks/data/firestore_user.dart";
+import "../../mocks/data/geopoint.dart";
+import "../../mocks/data/post_data.dart";
+
+Future<List<PostData>> addPosts(
+  PostRepositoryService postRepository,
+  GeoPoint pos,
+  int n,
+) async {
+  final fakePosts = PostDataGenerator.generatePostData(n);
+  for (final post in fakePosts) {
+    await postRepository.addPost(post, pos);
+  }
+  return fakePosts;
+}
```

Comment:
I'm feeling this could have its place in other tests too. I think it could make sense to move it to a util file for tests. For instance, why not as a method to `PostDataGenerator`.

Reply:
I don't really like it in the end. I prefer what we have in the post_repository_test, with full post generation and then setting the database directly. We could maybe move the `setPostFirestore` and `setPostsFirestore` methods defined there to a util file instead, but I'm not sure where.

Reply:
I allow myself to insist, I don't think those functions have their place here. There is a `testutils/` package in which you can put such functions, but I'm feeling this would not be great to find those functions later.
I allow myself to insist, I think they have their place in the `PostDataGenerator`.

Reply:
Done in 4793ee6f55dbb6c2057b062d809c160ee89dd7b1 and 3a976865bc1ac79d87849f68e4950d28dea02bd4. I changed the implementation to be inside the post generator and to use setPost. That way the ids simply increase starting from 0, and that will make debugging failing tests easier. So I have also added some methods to the post generator and moved setPost from the post test to the generator file that way we can reuse it too.